### PR TITLE
Navigate to alternate page when changing language with fallback to home page

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Smarter navigation when changing language
- Do full reload on country change
- Try to find alternate in current language otherwise
- Fallback to home page

## Justify why they are needed

- Prevent 404 when changing language in nested pages
- Improve user experience

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1372** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1372" target="_blank">Graphite</a> 
    * **PR #1374** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1374" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1374?utm_source=stack-comment).

File: apps/store/src/blocks/FooterBlock.tsx [hunk 1]
```diff
@@ -62,14 +55,15 @@ export const FooterSection = ({ blok }: FooterSectionProps) => {
 }
 FooterSection.blockName = 'footerSection' as const
 
-export type FooterBlockProps = SbBaseBlockProps<{
+export type FooterBlockProps = {
+  onLocaleChange?: (newLocale: IsoLocale) => void
```
Comment (robinandeer) [inline apps/store/src/blocks/FooterBlock.tsx:59]: nit: maybe this should be required

**Review by robinandeer — APPROVED** [review]
Looks good!

**Comment by alebedev** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: XFNoKz4DNNvwwLMmn3Qx.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1374)

**Pull request merged**